pkg/image/qemu.go

```diff
@@ -131,6 +135,18 @@ func convertToRaw(src, dest string, preallocate bool) error {
 		return errors.Wrap(err, errorMsg)
 	}
 
+	// With writeback cache mode it's possible that the process will exit before all writes have been commited to storage.
+	// To guarantee that our write was commited to storage, we make a syncfs syscall and ensure success.
+	file, err := os.Open(dest)
+	if err != nil {
+		return errors.Wrap(err, "could not get file descriptor for syncfs call following qemu-img writing")
+	}
+	if err := unix.Syncfs(int(file.Fd())); err != nil {
```

Comment:
I read [the syncfs documentation](https://linux.die.net/man/2/syncfs) and I am not sure it does the right thing for a block device. Perhaps this should be fsync instead

Reply:
I should have probably added this in the PR description, but here is my reasoning for not going with fsync:
https://blogs.gnome.org/cneumair/2006/02/11/ioctl-fsync-how-to-flush-block-device-buffers/#comment-91
And
```
According to the standard specification (e.g., POSIX.1-2001),
       sync() schedules the writes, but may return before the actual
       writing is done.  However Linux waits for I/O completions, and
       thus sync() or syncfs() provide the same guarantees as fsync()
       called on every file in the system or filesystem respectively.
```
From https://man7.org/linux/man-pages/man2/sync.2.html "Notes" section

Reply:
So in the blog post the author had, say, /dev/sda mounted at /mnt.
And fsync(/dev/sda) did not sync the pending filesystem writes to /mnt.
This isn't the case for us, we don't have a filesystem mounted at the same time. We are writing directly to /dev/sda.

Also, just stating the same comment more clearly:
syncfs(/mnt/disk.img) will sync the entire /mnt filesystem - good
syncfs(/dev/sda) will sync the /dev filesystem - not the block device - bad

Reply:
Got it, thank you
Updated the PR to fsync